client/modules/IDE/components/AddToCollectionList.jsx

```diff
@@ -29,20 +30,14 @@ export const QuickAddWrapper = styled.div`
 
 const AddToCollectionList = ({ projectId }) => {
   const { t } = useTranslation();
-
   const dispatch = useDispatch();
-
   const username = useSelector((state) => state.user.username);
-
   const collections = useSelector(getSortedCollections);
-
-  // TODO: improve loading state
   const loading = useSelector((state) => state.loading);
-  const [hasLoadedData, setHasLoadedData] = useState(false);
-  const showLoader = loading && !hasLoadedData;
 
   useEffect(() => {
-    dispatch(getCollections(username)).then(() => setHasLoadedData(true));
+    dispatch(startLoader());
+    dispatch(getCollections(username)).then(() => dispatch(stopLoader()));
```

Comment:
Check out the source of the `getCollections` thunk.  I think that it dispatches `startLoader` and `stopLoader` on its own already?